### PR TITLE
#723 handle removed entity by registering a dummy entity

### PR DESCRIPTION
**Pull request by chengchen**
Hey @bartoszwalacik, I tried to make the tests pass but I am not sure about the implementation if it fits your standard, since I am not fully aware of the whole picture of javers.

### CODE REVIEW

**Comment by bartoszwalacik** [comment]
i will take a look, but looks like this issue is more complicated than I thought

**Comment by chengchen** [comment]
@bartoszwalacik After testing the integrations with our applications, it seems that it's not enough the spec... To retrieve the owner snapshots, it's still not working (but the changes should be similar). I am rejecting the PR.

File: javers-core/src/main/java/org/javers/core/metamodel/type/TypeMapper.java [hunk 1]
```diff
@@ -168,6 +170,33 @@ public ManagedType getJaversManagedType(GlobalId globalId){
         return getJaversManagedType(state.getClassByTypeName(typeName), expectedType);
     }
 
+    /**
+     * @throws JaversException TYPE_NAME_NOT_FOUND if given typeName is not registered
```
Comment (chengchen) [inline javers-core/src/main/java/org/javers/core/metamodel/type/TypeMapper.java]: TYPE_NAME_NOT_FOUND is the only exception here? If yes, then the comment needs to be refined.

**Comment by bartoszwalacik** [comment]
@chengchen finally, I have implemenented it myself, see https://github.com/javers/javers/pull/725
and I'd like to merge my version.
Your version works but `RemovedEntity.java` would be problematic. Your work was very valuable, it helped me to solve this problem.

You are encouraged to contribute more to javers.

**Comment by chengchen** [comment]
@bartoszwalacik no worries. As long as I helped :D
Care to tell why RemovedEntity is problematic here? I would like to understand as well.

**Comment by bartoszwalacik** [comment]
Javers uses information about property name and type to deserialize Snapshots. 
For example, if your class has Id field of type `int`, value of this field in deserialized Snapshot would be also `int` type. If this class is not found, JSON basic types are used, so the type of this value would be `double`. In this context, it's better to have no type info than wrong type info (like in fake RemovedEntity class).

**Pull request closed without merging**